Add cheatsheet for `pathway-analysis` module (#277)
AlexsLemonade/training-modules: module-cheatsheets/pathway-analysis-cheatsheet.md

| `msigdbr`| [`msigdbr()`](https://www.rdocumentation.org/packages/msigdbr/versions/7.1.1/topics/msigdbr)| geom vertical line| Retrieves the specified MSigDB dataset |

### AnnotationDbi 
Read the `AnnotationDbi` package vignette [**here**](http://bioconductor.org/packages/release/bioc/vignettes/AnnotationDbi/inst/doc/IntroToAnnotationPackages.pdf).

| Library/Package                                  | Piece of Code                               | What it's called                               | What it does                                                                                                                                   |
|-----------------------------------------|---------------------------------------------|-------------------------|------------------------------------------------------------------------|
| `AnnotationDbi`        | [`keytypes()`](https://bioconductor.org/packages/release/bioc/vignettes/AnnotationDbi/inst/doc/IntroToAnnotationPackages.pdf)      | Keytypes     | Returns a character vector of column names/types of gene identifiers (e.g. `ENSEMBL` ) available in an `AnnotationDbi` package.                                              |

> | `AnnotationDbi`       | [`mapIDs()`](https://www.rdocumentation.org/packages/AnnotationDbi/versions/1.44.0/topics/AnnotationDb-objects) | Mapped IDs       | Extracts the mapped IDs for a set of gene identifiers. The types of gene identifiers for mapping are supplied to the arguments column (type of gene identifiers to return e.g. `ENSEMBL`) and keytype (type of gene identifiers to start with e.g. `ENTREZ`).                                  |  

Reviewer: ```suggestion
| `AnnotationDbi`       | [`mapIDs()`](https://www.rdocumentation.org/packages/AnnotationDbi/versions/1.44.0/topics/AnnotationDb-objects) | Mapped IDs       | Extracts the mapped IDs for a set of gene identifiers. The types of gene identifiers (e.g. `ENSEMBL` or `ENTREZ`) are supplied to arguments: `keytype` (type of gene identifiers we are providing in the `keys` argument) and  `column` (type of gene identifiers we want returned).                                  |  
```

I get confused by this everytime still; so I think this is good to explain :+1: I've just rearranged the wording a tad.

Reviewer 2: `mapIDs()` -> `mapIds()`

Reviewer: They used to have it the other way didn't they? I think that's why I'm getting confused.

Reviewer 2: Sorry @cansavvy I only read this comment in email and did not notice that you caught the same thing!

Reviewer: I didn't catch it. I edited the other part of the comment - missing that that was wrong. But then after you commented I  edited my suggestion.

Reviewer 2: 🤯

| `enrichplot`                | [`dotplot()`](https://yulab-smu.github.io/clusterProfiler-book/chapter12.html#dot-plot)             | Dot plot                                         | Produces a dot plot for given enrichment results                         |
| `enrichplot`                | [`upsetplot()`](https://yulab-smu.github.io/clusterProfiler-book/chapter12.html#upset-plot)             | Upset plot                                       | Produces an upset plot, which shows the overlapping genes between gene sets, for given enrichment results                      |
| `enrichplot`                | [`gseaplot()`](https://yulab-smu.github.io/clusterProfiler-book/chapter12.html#running-score-and-preranked-list-of-gsea-result)             | GSEA plot                                       | Produces a plot visualization displaying the distribution of gene set and enrichment score                      |

### `clusterProfiler`

Read the `clusterProfiler` package documentation [**here**](https://www.rdocumentation.org/packages/clusterProfiler/versions/3.0.4).

> | Library/Package               | Piece of Code                                                 | What it's called      | What it does                                                             |

Reviewer 2: We need `enricher()` here as well

<div style="page-break-after: always;"></div>

### `msigdbr`

Read the `msigdbr` documentation [**here**](https://www.rdocumentation.org/packages/msigdbr/versions/7.1.1).

| Library/Package| Piece of Code| What it's called| What it does |
|----------------|--------------|-----------------|--------------|
| `msigdbr`| [`msigdbr()`](https://www.rdocumentation.org/packages/msigdbr/versions/7.1.1/topics/msigdbr)| geom vertical line| Retrieves the specified MSigDB dataset |

### AnnotationDbi 
Read the `AnnotationDbi` package vignette [**here**](http://bioconductor.org/packages/release/bioc/vignettes/AnnotationDbi/inst/doc/IntroToAnnotationPackages.pdf).

| Library/Package                                  | Piece of Code                               | What it's called                               | What it does                                                                                                                                   |
|-----------------------------------------|---------------------------------------------|-------------------------|------------------------------------------------------------------------|

> | `AnnotationDbi`        | [`keytypes()`](https://www.rdocumentation.org/packages/AnnotationDbi/versions/1.44.0/topics/AnnotationDb-objects)      | Keytypes     | Returns a character vector of column names/keytypes (eg., type of gene identifiers) available in an `AnnotationDbi` package.                                              |

Reviewer: Are there no specific documentations for `keytypes()` I'm not seeing any in my Google searches. If so, I would probably make this link to the [vignette where these items](https://bioconductor.org/packages/release/bioc/vignettes/AnnotationDbi/inst/doc/IntroToAnnotationPackages.pdf)are explained instead of the general `AnnotationDbi`.

> | `AnnotationDbi`        | [`keytypes()`](https://www.rdocumentation.org/packages/AnnotationDbi/versions/1.44.0/topics/AnnotationDb-objects)      | Keytypes     | Returns a character vector of column names/keytypes (eg., type of gene identifiers) available in an `AnnotationDbi` package.                                              |

Reviewer: ```suggestion
| `AnnotationDbi`        | [`keytypes()`](https://www.rdocumentation.org/packages/AnnotationDbi/versions/1.44.0/topics/AnnotationDb-objects)      | Keytypes     | Returns a character vector of column names/keytypes, _i.e._ types of gene identifiers like `ENSEMBL` available in an `AnnotationDbi` package.                                              |
```

> | `AnnotationDbi`       | [`mapIDs()`](https://www.rdocumentation.org/packages/AnnotationDbi/versions/1.44.0/topics/AnnotationDb-objects) | Mapped IDs       | Extracts the mapped ids for a set of keys (e.g., gene identifiers) of a specific keytype                                    |  

Reviewer: The words `keys`, `keytypes`, and `columns` are kinda ambiguous/confusing for this. Can we put this in more plain language? May want to reference [Jackie's wording about mapIds here](https://github.com/AlexsLemonade/training-modules/blob/4e191a2fb621f6826299b5d02c8566d3dd6eb484/pathway-analysis/01-overrepresentation_analysis.Rmd#L161).

### Base `R`

Read the Base `R` documentation [**here**](https://www.rdocumentation.org/packages/base/versions/3.5.1).

|Library/Package|Piece of Code|What it's called| What it does|
|---------------|-------------|----------------|-------------|
| Base `R`| [`fisher.test()`](https://www.rdocumentation.org/packages/stats/versions/3.6.2/topics/fisher.test)| Fisher's Exact Test | Performs the Fisher's exact test for testing the null of independence of rows and columns for a given matrix or data.frame with count data |

<div style="page-break-after: always;"></div>

### `DESeq2`

> Read the `DESeq2` package documentation [**here**](https://bioc.ism.ac.jp/packages/3.8/bioc/manuals/DESeq2/man/DESeq2.pdf), and the package vignette by Love, Anders, and Huber [**here**](https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html).

Reviewer: ```suggestion
Read the `DESeq2` package documentation [**here**](https://bioc.ism.ac.jp/packages/3.8/bioc/manuals/DESeq2/man/DESeq2.pdf) and the package vignette by Love, Anders, and Huber [**here**](https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html).
```

| `enrichplot`                | [`gseaplot()`](https://yulab-smu.github.io/clusterProfiler-book/chapter12.html#running-score-and-preranked-list-of-gsea-result)             | GSEA plot                                       | Produces a plot visualization displaying the distribution of gene set and enrichment score                      |

### `clusterProfiler`

Read the `clusterProfiler` package documentation [**here**](https://www.rdocumentation.org/packages/clusterProfiler/versions/3.0.4).

| Library/Package               | Piece of Code                                                 | What it's called      | What it does                                                             |
|-------------------------------|--------------------------------------------------------------|--------------------------------|--------------------------------------------------------------------------|

> | `clusterProfiler`                | [`GSEA()`](https://www.rdocumentation.org/packages/clusterProfiler/versions/3.0.4/topics/GSEA)             | Gene Set Enrichment Analysis (GSEA)                                         | Performs a universal gene set enrichment analysis on given set of gene sets                       |

Reviewer: What do the gene sets provided need to look like in R object terms?

Author: I'll put more detail around that 👍

> | `clusterProfiler`                | [`enricher()`](https://www.rdocumentation.org/packages/clusterProfiler/versions/3.0.4/topics/enricher)             | Enricher                                         | Performs a universal over-represention/enrichment analysis for a given genes of interest list                  |

Author: How's this description for `enricher()` @jaclyn-taroni?

> | `clusterProfiler`                | [`enricher()`](https://www.rdocumentation.org/packages/clusterProfiler/versions/3.0.4/topics/enricher)             | Enricher                                         | Performs a universal over-represention/enrichment analysis for a given genes of interest list                  |

Reviewer 2: ```suggestion
| `clusterProfiler`                | [`enricher()`](https://www.rdocumentation.org/packages/clusterProfiler/versions/3.0.4/topics/enricher)             | Enricher                                         | Performs a universal over-representation analysis for a given list of genes and gene sets or pathways             |
```

<div style="page-break-after: always;"></div>

### `GSVA`

Read the `GSVA` package documentation [**here**](https://www.rdocumentation.org/packages/GSVA/versions/1.20.0).

| Library/Package               | Piece of Code                                                 | What it's called      | What it does                                                             |
|-------------------------------|--------------------------------------------------------------|--------------------------------|--------------------------------------------------------------------------|
| `GSVA`                | [`gsva()`](https://www.rdocumentation.org/packages/GSVA/versions/1.20.0/topics/gsva)             | Gene Set Variation Analysis (GSVA)                                         | Estimates gene set variation analysis enrichment scores on given gene expression matrix                       |

### `qusage`

Read the `qusage` package documentation [**here**](https://www.rdocumentation.org/packages/qusage/versions/2.4.0).

| Library/Package|Piece of Code| What it's called| What it does  |
|----------------|-------------|-----------------|---------------|

> | `qusage` | [`read.gmt)`](https://www.rdocumentation.org/packages/qusage/versions/2.4.0/topics/read.gmt) | Read in `.gmt` files | Reads in gene set information from `.gmt` files |

Reviewer: ```suggestion
| `qusage` | [`read.gmt()`](https://www.rdocumentation.org/packages/qusage/versions/2.4.0/topics/read.gmt) | Read in `.gmt` files | Reads in gene set information from `.gmt` files |
```

Author: Good catch!
